src/webrtc.cpp

```diff
@@ -37,9 +37,19 @@ static void oai_onconnectionstatechange_task(PeerConnectionState state,
 #endif
   } else if (state == PEER_CONNECTION_CONNECTED) {
 #ifndef LINUX_BUILD
+#if CONFIG_SPIRAM
+    constexpr size_t stack_size = 20000;
     StackType_t *stack_memory = (StackType_t *)heap_caps_malloc(
         20000 * sizeof(StackType_t), MALLOC_CAP_SPIRAM);
-    xTaskCreateStaticPinnedToCore(oai_send_audio_task, "audio_publisher", 20000,
+#else // CONFIG_SPIRAM
+    constexpr size_t stack_size = 20000;
+    StackType_t *stack_memory = (StackType_t *)malloc(stack_size * sizeof(StackType_t));
+#endif // CONFIG_SPIRAM
```

Comment:
Just a few notes about this code from a passer-by:

`CONFIG_SPIRAM=y` is not a sufficient condition to be sure that we can allocate from SPIRAM. `CONFIG_SPIRAM_IGNORE_NOTFOUND` might be set, in which case the application will boot up and ignore the lack of SPI RAM. `SPIRAM_USE_MEMMAP` might also be set, in which case SPIRAM will initialized but won't be available for `malloc` or `heap_caps_malloc`.

We can use `heap_caps_malloc_prefer`, instead, and ask it to use SPIRAM and fall back to the internal RAM otherwise. 


```c++
constexpr size_t stack_size = 20000;
StackType_t *stack_memory = (StackType_t *)heap_caps_malloc_prefer(
        20000 * sizeof(StackType_t), 2, MALLOC_CAP_SPIRAM | MALLOC_CAP_8BIT, MALLOC_CAP_INTERNAL|MALLOC_CAP_8BIT);
```